Clarify Finalizer Usage and Controller Types in Background Cascading Deletion (#47977)
kubernetes/website: content/en/docs/concepts/architecture/garbage-collection.md

### Background cascading deletion {#background-deletion}

In background cascading deletion, the Kubernetes API server deletes the owner

> object immediately and the controller (like the custom or default controller)

Reviewer: Does this controller have a name? Default controller is a bit vague 

Thank you for the pr

Author: I believe this is the [garbage collector](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/graph_builder.go) controller.

Author: I could use the below instead.
```suggestion
object immediately and the garbage collector controller (custom or default)
```

> object immediately and the controller (like the custom or default controller)
> cleans up the dependent objects in the background.

Reviewer 2: Perhaps it's better to remove "(like the custom or default controller)" to avoid confusion
```suggestion
object immediately and the controller cleans up the dependent objects in the background.
```

Author: I included it due to a complaint in the referenced issue. The issue creator was using a custom controller and wasn’t sure if the custom one was also supported. Kindly see this [review comment](https://github.com/kubernetes/website/pull/47977/files#r1766248775) where I specified the name of the controller.

> If a finalizer exists, it ensures that objects are not deleted until all necessary clean-up tasks are completed.

Reviewer 2: This may be better

```suggestion
If a finalizer exists, it ensures the resources the deleted object owned are deleted before the object itself.
```

Author: Thanks for the suggestion, @reylejano. The suggestion confused me at first due to the duplicated “the” used at proximity. I’ll paraphrase it to “`If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.`”

Author: ```suggestion
If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.
```

Reviewer 3: A finalizer is a blocker and the list of finalizers is mutable by other objects which can be controller or something else. The finalizer is not necessarily linked to an owner.

Reviewer 3: if a finalizer doesn't have to imply an ownership relation, isn't the revised description below problematic?

> If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.

Author: It is. Maybe we could use the initial sentence I suggested to avoid confusion, as it's more general.

> If a finalizer exists, it ensures that objects are not deleted until all necessary clean-up tasks are completed.

> If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.

Author: ```suggestion
If a finalizer exists, it ensures that objects are not deleted until all necessary clean-up tasks are completed.
```
